Ch10_Recurrent_Neural_Networks/Machine_Translation_and_Data_Sets.ipynb

```diff
@@ -0,0 +1,1198 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Machine Translation and Data Sets"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Machine translation (MT) refers to the automatic translation of a segment of text from one language to another. Solving this problem with neural networks is often called neural machine translation (NMT).Compared to the language model we discussed before, a major difference for MT is that the output is a sequence of words instead of a single words. The length of the output sequence could be different to the source sequence length. In the rest of this section, we will demonstrate how to pre-process a MT dataset and transform it into a set of data batches."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {
+    "colab": {},
+    "colab_type": "code",
+    "id": "OmMeMHFNr_aE"
+   },
+   "outputs": [],
+   "source": [
+    "import sys\n",
+    "sys.path.insert(0, '..')\n",
+    "import collections\n",
+    "import d2l\n",
+    "import zipfile\n",
+    "from d2l.data.base import Vocab\n",
+    "\n",
+    "import torch\n",
+    "import torch.nn as nn\n",
+    "import torch.nn.functional as F\n",
+    "from torch.utils import data as gdata"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Read and Pre-process Data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We first download a dataset that contains a set of English sentences with the corresponding French trans-\n",
+    "lations. As can be seen that each line contains a English sentence with its French translation, which are\n",
+    "separated by a TAB."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/",
+     "height": 347
+    },
+    "colab_type": "code",
+    "id": "HfPtS21avd-t",
+    "outputId": "691fda1c-1f86-466e-bf49-22d31c095ae6"
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "--2019-06-08 15:18:12--  http://www.manythings.org/anki/fra-eng.zip\n",
```

Comment:
Here make two different code blocks. 
One to `wget and reading the zip` data, and,
Other to print out `print(raw_text[0:95])`